providers/implementations/signature/dsa_sig.c

```diff
@@ -131,11 +131,11 @@ static int dsa_setup_md(PROV_DSA_CTX *ctx,
                                                             sha1_allowed);
         size_t mdname_len = strlen(mdname);
 
-        if (md == NULL || md_nid == NID_undef) {
```

Comment:
Should this be `<=` rather than `<` ?

Reply:
Sure.  I wasn't completely sure and the old code checked for NID_undef which is zero, so better to verify that this was the intention.